main.cpp

```diff
@@ -1,479 +1,535 @@
-            Brightness (double)
-            width in cm (int)
-            height in cm (int)
-
-Step 3 is written below UDTs 5 - 9
-*/
+struct Knife
+{
+    // length
+    float length = 3.0f;
+    // sharpness
+    float sharpness = 0.8f;
+    // mass
+    float mass = 0.50f;
+    // material
+    std::string material = "Stainless Steel";
+    // reflectivity
+    float reflectivity = 0.8f;
+
+    void cut();
+    void scratch();
+    void whittle();
+    std::string getMaterial();
+    void setMaterial(std::string newMaterial);
+};
 
 /*
-Thing 5)
+Thing 6) Nail clippers
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) length (float)
+    2) material (std::string)
+    3) clipper mouth width (float)
+    4) clipper mouth curve radius (float)
+    5) clipper strength (float)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) clip nail
+    2) unfold
+    3) fold
  */
 
+struct NailClippers
+{
+    // length
+    float length = 3.0f;
+    // material
+    std::string material = "Stainless Steel";
+    // mouth width
+    float mouthWidth = 0.12f;
+    // mouth curve radius
+    float mouthCurveRadius = 0.65f;
+    // clip strength
+    float clipStrength = 24.5f;
+
+    bool canClip(float requiredClipStrength = 20.f); // returns whether these clippers' clip strength meets or exceeds the required clip strength
+};
+
 /*
-Thing 6)
+Thing 7) Magnifying glass
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) lens major radius (float)
+    2) lens light refraction angle (float)
+    3) lens magnification factor (float)
+    4) arm material (std::string)
+    5) glass type (std::string)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) magnify image
+    2) focus light
+    3) flip image
  */
 
+struct MagnifyingGlass
+{
+    // lens major radius
+    float lensRadius = 1.f;
+    // lens light refraction angle
+    float refractionAngle = 30.f;
+    // lens magnification factor
+    float magFactor = 2.5f;
+    // arm material
+    std::string armMaterial = "Stainless Steel";
+    // glass type
+    std::string glassType = "Clear";
+
+    void magnifyImage(float magnificationFactor = 2.f);
+    void focusLight();
+    void flipImage();
+};
+
 /*
-Thing 7)
+Thing 8) Hair comb
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) length (float)
+    2) material (std::string)
+    3) number of teeth (int)
+    4) teeth pitch (float)
+    5) teeth length (float)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) comb hair
+    2) unfold
+    3) fold
  */
 
+struct HairComb
+{
+    // length
+    float length = 5.5f;
+    // material
+    std::string material = "Aluminum";
+    // number of teeth
+    int numTeeth = 30;
+    // teeth pitch
+    float teethPitch = .0125f;
+    // teeth length
+    float teethLength = 1.125f;
+
+    void combHair(int numStrokes = 1);
+    void breakTeeth(int numTeethToBreak = 1);
+    double getTotalTeethLength(float numTeeth, float pitch);
```

Comment:
Some of these functions don't match the plain-english descriptions.
either revise the functions or change the plain-english.
- [ ] revise